From 7e0f993c89113500144f8b68ed96dd160f37487e Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Fri, 4 May 2018 17:30:14 +0800
Subject: [PATCH] dts: rockchip: Set pwm pin pull down when used for negative
 pwm regulator
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

As a second global reset, the GRF is not reset, the iomux and
pull of PWM pin is still keeping, but PWM controller is reset,
PWM pin goes into input mode. However, the pull is still none
changed in kernel, which can cause voltage problems, so should
always keep the PWM pin pull down mode, with 0~50 Î¼A power
increase.

Change-Id: Ibbb9465f7c550d49d416bc3438c5199434df6eba
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 arch/arm/boot/dts/rk3128h-box.dts             |  2 ++
 arch/arm/boot/dts/rk3128x.dtsi                | 16 ++++++++++++
 arch/arm/boot/dts/rk3229-evb.dts              |  4 +++
 arch/arm/boot/dts/rk322x.dtsi                 | 16 ++++++++++++
 .../boot/dts/rk3288-evb-android-rk808-edp.dts |  2 ++
 arch/arm/boot/dts/rk3288.dtsi                 | 17 +++++++++++++
 .../boot/dts/rockchip/rk3308-evb-v10.dtsi     |  2 ++
 arch/arm64/boot/dts/rockchip/rk3308.dtsi      | 20 +++++++++++++++
 .../boot/dts/rockchip/rk3368-r88-dcdc.dts     |  2 ++
 arch/arm64/boot/dts/rockchip/rk3368.dtsi      | 12 +++++++++
 .../rockchip/rk3399-box-808-android-6.0.dts   |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399-box.dtsi  |  2 ++
 .../boot/dts/rockchip/rk3399-evb-rev2.dtsi    |  2 ++
 .../dts/rockchip/rk3399-firefly-linux.dts     |  2 ++
 .../boot/dts/rockchip/rk3399-rv1-android.dts  |  2 ++
 .../boot/dts/rockchip/rk3399-sapphire.dtsi    |  2 ++
 arch/arm64/boot/dts/rockchip/rk3399.dtsi      | 25 +++++++++++++++++++
 17 files changed, 130 insertions(+)

diff --git a/arch/arm/boot/dts/rk3128h-box.dts b/arch/arm/boot/dts/rk3128h-box.dts
index b8cf29650bf6..1facaf6f3acc 100644
--- a/arch/arm/boot/dts/rk3128h-box.dts
+++ b/arch/arm/boot/dts/rk3128h-box.dts
@@ -401,6 +401,8 @@
 
 &pwm1 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm1_pin_pull_down>;
 };
 
 &pwm2 {
diff --git a/arch/arm/boot/dts/rk3128x.dtsi b/arch/arm/boot/dts/rk3128x.dtsi
index e98d5fcbfd18..3514e23d93f4 100644
--- a/arch/arm/boot/dts/rk3128x.dtsi
+++ b/arch/arm/boot/dts/rk3128x.dtsi
@@ -1545,24 +1545,40 @@
 			pwm0_pin: pwm0-pin {
 				rockchip,pins = <3 21 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm0_pin_pull_down: pwm0-pin-pull-down {
+				rockchip,pins = <3 21 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm1 {
 			pwm1_pin: pwm1-pin {
 				rockchip,pins = <0 30 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm1_pin_pull_down: pwm1-pin-pull-down {
+				rockchip,pins = <0 30 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		pwm2 {
 			pwm2_pin: pwm2-pin {
 				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm2_pin_pull_down: pwm2-pin-pull-down {
+				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3 {
 			pwm3_pin: pwm3-pin {
 				rockchip,pins = <1 11 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm3_pin_pull_down: pwm3-pin-pull-down {
+				rockchip,pins = <1 11 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		spdif {
diff --git a/arch/arm/boot/dts/rk3229-evb.dts b/arch/arm/boot/dts/rk3229-evb.dts
index 5cd2cc2d5c65..096840d116b6 100644
--- a/arch/arm/boot/dts/rk3229-evb.dts
+++ b/arch/arm/boot/dts/rk3229-evb.dts
@@ -327,10 +327,14 @@
 
 &pwm1 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm1_pin_pull_down>;
 };
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &sdio {
diff --git a/arch/arm/boot/dts/rk322x.dtsi b/arch/arm/boot/dts/rk322x.dtsi
index 7fd83f7fff31..e9db63bb97db 100644
--- a/arch/arm/boot/dts/rk322x.dtsi
+++ b/arch/arm/boot/dts/rk322x.dtsi
@@ -1559,24 +1559,40 @@
 			pwm0_pin: pwm0-pin {
 				rockchip,pins = <3 21 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm0_pin_pull_down: pwm0-pin-pull-down {
+				rockchip,pins = <3 21 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm1 {
 			pwm1_pin: pwm1-pin {
 				rockchip,pins = <0 30 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm1_pin_pull_down: pwm1-pin-pull-down {
+				rockchip,pins = <0 30 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		pwm2 {
 			pwm2_pin: pwm2-pin {
 				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm2_pin_pull_down: pwm2-pin-pull-down {
+				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3 {
 			pwm3_pin: pwm3-pin {
 				rockchip,pins = <1 11 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm3_pin_pull_down: pwm3-pin-pull-down {
+				rockchip,pins = <1 11 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		spdif {
diff --git a/arch/arm/boot/dts/rk3288-evb-android-rk808-edp.dts b/arch/arm/boot/dts/rk3288-evb-android-rk808-edp.dts
index 1e5561af30f4..90e7aa0fb9f7 100644
--- a/arch/arm/boot/dts/rk3288-evb-android-rk808-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk808-edp.dts
@@ -420,6 +420,8 @@
 
 &pwm1 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm1_pin_pull_down>;
 };
 
 &pinctrl {
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 6e40187a3eac..099470e60c4a 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -2293,24 +2293,41 @@
 			pwm0_pin: pwm0-pin {
 				rockchip,pins = <7 0 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm0_pin_pull_down: pwm0-pin-pull-down {
+				rockchip,pins = <7 0 RK_FUNC_1 &pcfg_pull_down>;
+			};
+
 		};
 
 		pwm1 {
 			pwm1_pin: pwm1-pin {
 				rockchip,pins = <7 1 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm1_pin_pull_down: pwm1-pin-pull-down {
+				rockchip,pins = <7 1 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm2 {
 			pwm2_pin: pwm2-pin {
 				rockchip,pins = <7 22 3 &pcfg_pull_none>;
 			};
+
+			pwm2_pin_pull_down: pwm2-pin-pull-down {
+				rockchip,pins = <7 22 3 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3 {
 			pwm3_pin: pwm3-pin {
 				rockchip,pins = <7 23 3 &pcfg_pull_none>;
 			};
+
+			pwm3_pin_pull_down: pwm3-pin-pull-down {
+				rockchip,pins = <7 23 3 &pcfg_pull_down>;
+			};
 		};
 
 		gmac {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index 5882595bb7dd..c86ac1ce8b37 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -707,6 +707,8 @@
 
 &pwm0 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm0_pin_pull_down>;
 };
 
 &u2phy {
diff --git a/arch/arm64/boot/dts/rockchip/rk3308.dtsi b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
index b4fadd15bd0a..cf7329f66159 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
@@ -1902,6 +1902,11 @@
 				rockchip,pins =
 					<0 RK_PB5 1 &pcfg_pull_none>;
 			};
+
+			pwm0_pin_pull_down: pwm0-pin-pull-down {
+				rockchip,pins =
+					<0 RK_PB5 1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm1 {
@@ -1909,6 +1914,11 @@
 				rockchip,pins =
 					<0 RK_PB6 1 &pcfg_pull_none>;
 			};
+
+			pwm1_pin_pull_down: pwm1-pin-pull-down {
+				rockchip,pins =
+					<0 RK_PB6 1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm2 {
@@ -1916,6 +1926,11 @@
 				rockchip,pins =
 					<0 RK_PB7 1 &pcfg_pull_none>;
 			};
+
+			pwm2_pin_pull_down: pwm2-pin-pull-down {
+				rockchip,pins =
+					<0 RK_PB7 1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3 {
@@ -1923,6 +1938,11 @@
 				rockchip,pins =
 					<0 RK_PC0 1 &pcfg_pull_none>;
 			};
+
+			pwm3_pin_pull_down: pwm3-pin-pull-down {
+				rockchip,pins =
+					<0 RK_PC0 1 &pcfg_pull_down>;
+			};
 		};
 
 		gmac {
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-r88-dcdc.dts b/arch/arm64/boot/dts/rockchip/rk3368-r88-dcdc.dts
index 510ebdf9e670..f65123d3b86d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-r88-dcdc.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-r88-dcdc.dts
@@ -242,6 +242,8 @@
 
 &pwm1 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm1_pin_pull_down>;
 };
 
 &i2c0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3368.dtsi b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
index 89074a8d0941..a880a393bf0c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
@@ -2029,6 +2029,10 @@
 				rockchip,pins = <3 8 RK_FUNC_2 &pcfg_pull_none>;
 			};
 
+			pwm0_pin_pull_down: pwm0-pin-pull-down {
+				rockchip,pins = <3 8 RK_FUNC_2 &pcfg_pull_down>;
+			};
+
 			vop_pwm_pin: vop-pwm {
 				rockchip,pins = <3 8 RK_FUNC_3 &pcfg_pull_none>;
 			};
@@ -2038,12 +2042,20 @@
 			pwm1_pin: pwm1-pin {
 				rockchip,pins = <0 8 RK_FUNC_2 &pcfg_pull_none>;
 			};
+
+			pwm1_pin_pull_down: pwm1-pin-pull-down {
+				rockchip,pins = <0 8 RK_FUNC_2 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3 {
 			pwm3_pin: pwm3-pin {
 				rockchip,pins = <3 30 RK_FUNC_3 &pcfg_pull_none>;
 			};
+
+			pwm3_pin_pull_down: pwm3-pin-pull-down {
+				rockchip,pins = <3 30 RK_FUNC_3 &pcfg_pull_down>;
+			};
 		};
 
 		sdio0 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box-808-android-6.0.dts b/arch/arm64/boot/dts/rockchip/rk3399-box-808-android-6.0.dts
index 8e853520558d..dce74b6fb09f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box-808-android-6.0.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box-808-android-6.0.dts
@@ -606,6 +606,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &pwm3 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
index a8b03e5b6128..206de93a89e6 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-box.dtsi
@@ -613,6 +613,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &pwm3 {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
index 2d2c14d05036..74210be834ce 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
@@ -337,6 +337,8 @@
 
 &pwm3 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm3a_pin_pull_down>;
 };
 
 &u2phy0_otg {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
index 8cb99aa904f2..beffd8dbb219 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
@@ -850,6 +850,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &rkvdec {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts
index 1471ca997cdd..10ad4de78c05 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rv1-android.dts
@@ -604,6 +604,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &rk_screen {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
index 2e0caf1e76a4..e2fc51bd3007 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire.dtsi
@@ -540,6 +540,8 @@
 
 &pwm2 {
 	status = "okay";
+	pinctrl-names = "active";
+	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
 &sdhci {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 84f38ab45019..eccccd847fe7 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -2801,6 +2801,11 @@
 					<4 18 RK_FUNC_1 &pcfg_pull_none>;
 			};
 
+			pwm0_pin_pull_down: pwm0-pin-pull-down {
+				rockchip,pins =
+					<4 18 RK_FUNC_1 &pcfg_pull_down>;
+			};
+
 			vop0_pwm_pin: vop0-pwm-pin {
 				rockchip,pins =
 					<4 18 RK_FUNC_2 &pcfg_pull_none>;
@@ -2817,6 +2822,11 @@
 				rockchip,pins =
 					<4 22 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm1_pin_pull_down: pwm1-pin-pull-down {
+				rockchip,pins =
+					<4 22 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm2 {
@@ -2824,6 +2834,11 @@
 				rockchip,pins =
 					<1 19 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm2_pin_pull_down: pwm2-pin-pull-down {
+				rockchip,pins =
+					<1 19 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3a {
@@ -2831,6 +2846,11 @@
 				rockchip,pins =
 					<0 6 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm3a_pin_pull_down: pwm3a-pin-pull-down {
+				rockchip,pins =
+					<0 6 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		pwm3b {
@@ -2838,6 +2858,11 @@
 				rockchip,pins =
 					<1 14 RK_FUNC_1 &pcfg_pull_none>;
 			};
+
+			pwm3b_pin_pull_down: pwm3b-pin-pull-down {
+				rockchip,pins =
+					<1 14 RK_FUNC_1 &pcfg_pull_down>;
+			};
 		};
 
 		edp {
-- 
2.35.3

